docs/backend/behaviors.md

```diff
@@ -11,12 +11,320 @@ myst:
+    )
+    price_vat = Attribute("VAT 20% of net price")
+    price_gross = Attribute("Price gross (net + VAT 20%")
+
+
+class IPriceMarker(Interface):
+    """Marker for content that has a price."""
+
+
+@implementer(IPriceBehavior)
+class PriceAdapter:
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def price_net(self):
+        """Getter, read from context and return back"""
+        return self.context.price_net
+
+    @price_net.setter
+    def price_net(self, value):
+        """Setter, called by the form, set on context"""
+        self.context.price_net = value
+
+    @property
+    def price_vat(self):
+        return self.price_net * 0.2
+
+    @property
+    def price_gross(self):
+        return self.price_net + self.price_vat
+
+```
+
+The registration in the `configure.zcml` looks like so:
+
+```XML
+  <plone:behavior
+      factory=".price.PriceAdapter"
+      for=".price.IPriceMarker"
+      marker=".price.IPriceMarker"
+      name="myproject.price"
+      provides=".price.IPriceBehavior"
+      title="Price with net, VAT and gross"
+  />
+  ```
+
+After a restart of Plone, the behavior can be added to the content type in the {guilabel}`Content Types` control panel.
+The add and edit forms contain a new field `Price (net)`.
+
+This field is not displayed in most views.
+To display the data entered in this field, you need to modify the page template by adding the `price_net` field as `context.price_net`.
+To access the `price_vat` and `price_gross` fields from a browser view, you need to get the adapter from the context of the view:
+
+```python
+from .price import IPriceBehavior
+
+class SomeViewClass:
+
+    def vat(self):
+        adapter = IPriceBehavior(context)
+        return adapter.price_vat
+
+    def gross(self):
+        adapter = IPriceBehavior(context)
+        return adapter.price_gross
+```
+
+### Creating a behavior with PloneCLI
+
+To add a behavior to your add-on, you can use PloneCLI as follows:
+
+```shell
+plonecli add behavior
+```
+
+This will create the behavior Python file in the `behaviors` folder, where you can define your behaviors schema fields, and registers the behavior in the `configure.zcml`.
+
+### Further reading on working with behaviors
 
 ```{seealso}
 See the chapter {ref}`training:behaviors1-label` from the Mastering Plone 6 Training.
 ```
 
-```{todo}
-Contribute to this documentation!
-See issue [Backend > Behaviors needs content](https://github.com/plone/documentation/issues/1302).
-```
\ No newline at end of file
+## How behaviors work
+
+```{note}
+Skip this section if you do not want to dive deeper into the internals of behaviors.
+You do not *need* to know this, but it may help if you run into problems.
+```
+
+In Plone, behaviors can be globally enabled on content types at runtime.
+With add-ons, behaviors can even be enabled even on a single content object or for a whole subtree in the content hierarchy.
+
+### Interfaces and adapters
+
```

Comment:
```suggestion
To explain interfaces and adapters, let's begin with an analogy using electrical systems.

An electrical outlet provides an interface through which electricity passes.
When you travel to another country, you may need an outlet adapter for the outlet (the interface).
For example, assume you have a device that has plug for Schuko outlets, and in Italy there are Type L outlets.
If we were to represent the behavior of choosing the correct power adapter in Plone, you would do the following.

-   Call `getAdapter(context, ITypeL)` to determine the type of adapter needed for the electric outlet's interface.
-   Call `getAdapter(context, ISchuko) to determine the type of adapter that will accept your device's plug.
-   You choose the correct electrical adapter to use, and plug that into the wall outlet.
    This is similar to calling an abstract factory which returns the correct adapter to use.
    This interface provides an ISchuko adapter.
-   Finally to use the new interface, you plug in your Schuko plug into the adapter to get power to your device.

Similarly, using the {ref}`behavior code example <behavior-code-example>` above:

-   You would call an abstract factory with `getAdapter(context, IPriceBehavior)` to get an adapter.
    Although it is an interface, it is more of a shortcut to factory usage.
-   The adapter that is specific to the given content type is assigned to the variable `adapter`.
    Now you can use `adapter` for whatever you like.

```

Reply:
It is possible that I might have made some mistakes in terms, so please correct them as needed.

Reply:
It is not correct in details, but great anyway, I will refine this. but at the moment I have a x-mas overload here. Please stay tuned.

Reply:
I commit this and propose a new change, this is probably easier and more transparent than editing yours